tool/main.re

```diff
@@ -246,22 +250,188 @@ static MULTIPLAYER_MENU = Ui::new("Multiplayer:", List::of(
     }),
 ));
 
+static mut RECORDING_NAME_LABEL = Text { text: "Recording name" };
+
+fn recording_operations(operation: string) {
+    RECORDING_NAME_LABEL.text = f"Recording name";
+    let recordings_list = Tas::list_recordings();
+    let mut recordings = List::of(
+        UiElement::Button(UiButton {
+            label: Text { text: "Back" },
+            onclick: fn(label: Text) { leave_ui() },
+        }),
+        UiElement::Input(Input {
+            label: RECORDING_NAME_LABEL,
+            input: "",
+            onclick: fn(input: string) {
+                let recordings_list = Tas::list_recordings();
+                if input.len_utf8() == 0 {
+                    RECORDING_NAME_LABEL.text = f"Recording name (Error: empty name)";
+                    return;
+                }
+                if recordings_list.contains(input) && operation == "load" {
+                    RECORDING_NAME_LABEL.text = f"Recording name";
+                    set_current_component(TAS_COMPONENT);
+                    leave_ui();
+                    leave_ui();
+                    leave_ui();
+                } else {
+                    RECORDING_NAME_LABEL.text = f"Recording name (Error: no such recording)";
+                }
+                if recordings_list.contains(input) && operation == "delete" {
```

Comment:
Instead of using string-based operations, create an enum defining all possible operations.